kubectl/README.md

```diff
@@ -0,0 +1,51 @@
+# Kubectl
+
+This Task deploys (or delete) a Kubernates resource (pod). It uses
+[`kubectl`](https://kubernetes.io/zh/docs/reference/kubectl/kubectl/) for that.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/kubectl/kubectl-deploy.yaml
+```
+
+## Install ClusterRolebinding
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/kubectl/clusterrolebinding.yaml
+```
+
+## Inputs 
+
+### Parameters
+
+* **resource:**: The content of the resource to deploy
```

Comment:
could this be called `manifest` to make it clear that this is not a name or reference but a configuration kind of YAML string?

also, in order to use this in the [KFP-Tekton compiler](https://github.com/kubeflow/kfp-tekton/issues/49) perspective, we would need the following parameters:

- `action`: `['get', 'create', 'apply', 'delete', 'replace', 'patch']` -- the action to perform to the resource
- `merge_strategy`: `['strategic', 'merge', 'json']` -- the strategy used to merge a patch, defaults to "strategic"
- `success_condition`: a label selector expression which describes the `success` condition
- `failure_condition`: a label selector expression which describes the `failure` condition
- `manifest`: the kubernetes manifest

Even though not all of them will be supported yet, can we start out with at least `action` and `manifest` and document that there is a limitation for `action` for now?

I am worried that once this is getting used by a larger group of consumers, it will be hard to make substantial changes to parameters without breaking downstream pipelines

Reply:
Sorry for delay, involved in a tough project.
The function you mentioned is correct and needed for a complete `kubernete resource hendler` features, it's too complex to just with `script` and `yml`, I think it should be a pre-prepared image.

Whatever, I will dig into and check if i can found something.